### PR TITLE
Add two missing CA rules

### PR DESCRIPTION
**Pull request by gewarren**
[Preview link](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2218?branch=pr-en-us-20921).

### CODE REVIEW

File: docs/fundamentals/code-analysis/quality-rules/snippets/csharp/all-rules/ca2218.cs [hunk 1]
```diff
@@ -0,0 +1,179 @@
+﻿using System;
+
+namespace ca2218
```
Comment (mavasani) [inline docs/fundamentals/code-analysis/quality-rules/snippets/csharp/all-rules/ca2218.cs:3]: We no longer generate this warning for C# as the C# compiler generates a CS warning for it. These two rules are now VB only in the ported analyzers.

File: docs/fundamentals/code-analysis/style-rules/language-rules.md [hunk 2]
```diff
@@ -118,7 +118,7 @@ The style rules in this section are applicable to C# language only.
 
 ### Implicit and explicit types
 
-The style rules in this section concern the use of the [var](/dotnet/csharp/language-reference/keywords/var) keyword versus an explicit type in a variable declaration. This rule can be applied separately to built-in types, when the type is apparent, and elsewhere.
```
Comment (mavasani) [inline docs/fundamentals/code-analysis/style-rules/language-rules.md:121]: I am actually deleting all the sections in this file in https://github.com/dotnet/docs/pull/20892

Comment (mavasani) [reply docs/fundamentals/code-analysis/style-rules/language-rules.md:121]: That PR also updates these links to be relative paths.

File: docs/fundamentals/code-analysis/quality-rules/snippets/vb/all-rules/ca2224.vb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+            Dim pt As Point = CType(obj, Point)
+
+            Return X = pt.X AndAlso Y = pt.Y
+
+        End Function
+
+        Public Shared Operator =(ByVal pt1 As Point,
+                             ByVal pt2 As Point) As Boolean
+            Return Object.Equals(pt1, pt2)
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/snippets/vb/all-rules/ca2224.vb:78]: Please check this code. Does Object.Equals() then call Point's implementation of Equals()?

File: docs/fundamentals/code-analysis/quality-rules/snippets/vb/all-rules/ca2224.vb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        Public Property X As Integer
+        Public Property Y As Integer
+
+        Public Sub New(x As Integer, y As Integer)
+            Me.X = x
+            Me.Y = y
+        End Sub
+
+        Public Overrides Function GetHashCode() As Integer
+            Return X Or Y
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/snippets/vb/all-rules/ca2224.vb:18]: nit: Use HashCode.Combine

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

You can :shipit: when ready.

**Pull request merged**